spring-integration-core/src/main/java/org/springframework/integration/handler/DelayHandler.java

```diff
@@ -145,19 +139,6 @@ public void setDefaultDelay(long defaultDelay) {
 	}
 
 	/**
-	 * Specify the name of the header that should be checked for a delay period
-	 * (in milliseconds) or a Date to delay until. If this property is set, any
-	 * such header value will take precedence over this handler's default delay.
-	 * @deprecated in favor of {@link #delayExpression}
-	 *
-	 * @param delayHeaderName The name of the header.
-	 */
-	@Deprecated
-	public void setDelayHeaderName(String delayHeaderName) {
-		this.delayHeaderName = delayHeaderName;
-	}
-
-	/**
```

Comment:
We need to separate the deprecation removal - can't be cherry-picked. Let's do it in some other 4.1 only PR.